src/components/Pagination.js

```diff
@@ -0,0 +1,145 @@
+    return null;
+  }
+
+  const pages = getPages(props.currentRefinement, props.nbPages, 2);
+
+  function onPageClick(page) {
+    props.refine(page);
+    props.onClick();
+  }
+
+  return (
+    <div className="ais-Pagination">
+      <ul className="ais-Pagination-list">
+        {props.currentRefinement !== 1 && (
+          <li className="ais-Pagination-item ais-Pagination-item--previousPage">
+            <a
+              className="ais-Pagination-link"
+              title="Previous page"
+              href={props.createURL(props.currentRefinement - 1)}
+              onClick={(event) => {
+                event.preventDefault();
+                onPageClick(props.currentRefinement - 1);
+              }}
+            >
+              <svg width={10} height={10} viewBox="0 0 10 10">
+                <g
+                  fill="none"
+                  fillRule="evenodd"
+                  stroke="currentColor"
+                  strokeLinecap="round"
+                  strokeLinejoin="round"
+                  strokeWidth="1.143"
+                >
+                  <path d="M9 5H1M5 9L1 5l4-4" />
+                </g>
+              </svg>
+            </a>
+          </li>
+        )}
+
+        {pages.map((page) => (
+          <li
+            className={[
+              'ais-Pagination-item',
+              'ais-Pagination-item--page',
+              props.currentRefinement === page &&
+                'ais-Pagination-item--selected',
+            ]
+              .filter(Boolean)
+              .join(' ')}
+            key={page}
+          >
+            <a
+              className="ais-Pagination-link"
+              title={`Page ${page}`}
+              href={props.createURL(page)}
+              onClick={(event) => {
+                event.preventDefault();
+                onPageClick(page);
+              }}
+            >
+              {page}
+            </a>
+          </li>
+        ))}
+
+        {props.currentRefinement !== props.nbPages && (
+          <li className="ais-Pagination-item ais-Pagination-item--nextPage">
+            <a
+              className="ais-Pagination-link"
+              title="Next page"
+              href={props.createURL(props.currentRefinement + 1)}
+              onClick={(event) => {
+                event.preventDefault();
+                onPageClick(props.currentRefinement + 1);
+              }}
+            >
+              <svg width={10} height={10} viewBox="0 0 10 10">
+                <g
+                  fill="none"
+                  fillRule="evenodd"
+                  stroke="currentColor"
+                  strokeLinecap="round"
+                  strokeLinejoin="round"
+                  strokeWidth="1.143"
+                >
+                  <path d="M1 5h8M5 9l4-4-4-4" />
+                </g>
+              </svg>
+            </a>
+          </li>
+        )}
+      </ul>
+    </div>
+  );
+});
+
+function getPaddingLeft(currentPage, padding, maxPages, size) {
```

Comment:
The below methods are taken from the [orignal implementation](https://github.com/algolia/react-instantsearch/blob/b41bff237e536a3da7e69988b2e9678a0e44d7d3/packages/react-instantsearch-dom/src/components/Pagination.js#L11-L44).